algebird-core/src/main/scala/com/twitter/algebird/Field.scala

```diff
@@ -54,6 +54,14 @@ object FloatField extends Field[Float] {
     assertNotZero(r)
     l / r
   }
+  override def sum(t: TraversableOnce[Float]): Float = {
+    var sum = 0.0f
+    t.foreach(sum += _)
```

Comment:
Closure usage is far far cheaper than boxing. I guess its to do with inlining or new gen allocations. But boxing will show up orders of magnitude higher than it. But seems reasonable to eliminate it here